packages/orchestration/src/service.js

```diff
@@ -192,49 +192,44 @@ const prepareOrchestrationKit = (
          * @param {IBCConnectionID} hostConnectionId the counterparty
          *   connection_id
          * @param {IBCConnectionID} controllerConnectionId self connection_id
-         * @returns {Promise<IcaAccount>}
+         * @returns {Vow<IcaAccount>}
          */
         makeAccount(chainId, hostConnectionId, controllerConnectionId) {
           const remoteConnAddr = makeICAChannelAddress(
             hostConnectionId,
             controllerConnectionId,
           );
           const portAllocator = getPower(this.state.powers, 'portAllocator');
-          return when(
-            watch(
-              E(portAllocator).allocateICAControllerPort(),
-              this.facets.requestICAChannelWatcher,
-              {
-                chainId,
-                remoteConnAddr,
-              },
-            ),
+          return watch(
+            E(portAllocator).allocateICAControllerPort(),
+            this.facets.requestICAChannelWatcher,
+            {
+              chainId,
+              remoteConnAddr,
+            },
           );
         },
         /**
          * @param {IBCConnectionID} controllerConnectionId
-         * @returns {Promise<ICQConnection>}
+         * @returns {Vow<ICQConnection> | ICQConnection}
```

Comment:
Do we need a reusable pattern for `Vow<T> | T`?  Are there other places where we should have that rather than simply `Vow<T>`?

Reply:
This was the only one I came across so far, but agree its worth considering